akirachix/Upile-Informational
# Staging

**What this PR does?**
**What is this?**
This PR introduces a CI/CD pipeline for the Next.js project. It automates the process of building, testing, and deploying the application, ensuring consistency and reducing manual intervention.
**Why this?**
Setting up a CI/CD pipeline ensures code quality and streamlines the deployment process. This reduces the chances of human error, speeds up development workflows, and enhances collaboration by automating key steps like testing and deployment.
**How are you doing this?**
   Added configuration for CI/CD using [specific tools used, e.g., GitHub Actions, Jenkins, CircleCI, etc.]


   Configured automatic builds on push and pull requests for the main branch.


   Integrated unit tests and code linting as part of the pipeline to ensure code quality.


   Set up deployment to [target environment, e.g., production, staging, etc.] after successful builds.


**List of updates:**
Added CI configuration file (e.g., .github/workflows/ci.yml or Jenkinsfile).
Configured automatic testing and linting.
Set up deployment steps for [target environment].
Integrated build status notifications (if applicable).


**Type of change**
 Optimization
 Improvement
**How can it be tested?**
Ensure the pipeline runs automatically on pushing new changes to the repository.
Confirm that the build passes successfully by reviewing the logs in the CI tool.
Verify that the tests and linting steps are executed as part of the pipeline.
Check that the application is deployed to the appropriate environment once the pipeline completes.
Where is it documented?
https://wambuinatalygakenye.atlassian.net/browse/UP-138?atlOrigin=eyJpIjoiYWM1NjQzMzEwODUyNGYyNDk0ZWYwMDYxZjQzYTY4MWUiLCJwIjoiaiJ9
